saos-search/src/main/java/pl/edu/icm/saos/search/analysis/solr/SeriesGenerator.java

```diff
@@ -28,29 +46,24 @@
     public Series<Object, Integer> generateSeries(JudgmentSeriesCriteria judgmentSeriesCriteria, XSettings xsettings) {
         
         JudgmentCriteria judgmentCriteria = judgmentSeriesCriteriaConverter.convert(judgmentSeriesCriteria);
+         
+        SolrQuery query = judgmentSearchQueryFactory.createQuery(judgmentCriteria, new Paging(0, 0));
         
-        // generic: create base query part 
+        xSettingsFacetQueryApplier.applyXSettingsToQuery(query, xsettings);
         
-        // specific: create facet query part
-        
-        // generic: execute query
+        QueryResponse response = null;
+        try {
+            response = solrServer.query(query);
+        } catch (SolrServerException e) {
+            log.warn("Error in generating series", e);
+            return new Series<Object, Integer>();
```

Comment:
It's not best to do it. If we get an exception (that normally shouldn't be the case) then we should rethrow it up as a runtime exception and handle it in webapp. Warn gives us nothing and returning an empty series is misleading to the user which can think that just their query resulted in empty response.